pennylane/measurements/vn_entanglement_entropy.py

```diff
@@ -0,0 +1,163 @@
+        )
+    return VnEntanglementEntropyMP(wires=(wires0, wires1), log_base=log_base)
+
+
+class VnEntanglementEntropyMP(StateMeasurement):
+    """Measurement process that computes the Von Neumann entanglement entropy between the provided wires.
+    Please refer to :func:`~pennylane.vn_entanglement_entropy` for detailed documentation.
+    Args:
+        wires (Sequence[.Wires]): The wires the measurement process applies to.
+        id (str): custom label given to a measurement instance, can be useful for some applications
+            where the instance has to be identified
+        log_base (float): base for the logarithm
+    """
+
+    def _flatten(self):
+        metadata = (("wires", tuple(self.raw_wires)), ("log_base", self.log_base))
+        return (None, None), metadata
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        wires: Optional[Sequence[Wires]] = None,
+        id: Optional[str] = None,
+        log_base: Optional[float] = None,
+    ):
+        self.log_base = log_base
+        super().__init__(wires=wires, id=id)
+
+    # pylint: disable=arguments-differ
+    @classmethod
+    def _primitive_bind_call(cls, wires: Sequence, **kwargs):
+        if cls._wires_primitive is None:  # pragma: no cover
+            # just a safety check
+            return type.__call__(cls, wires=wires, **kwargs)  # pragma: no cover
+        return cls._wires_primitive.bind(*wires[0], *wires[1], n_wires0=len(wires[0]), **kwargs)
+
+    def __repr__(self):
+        return f"VnEntanglementEntropy(wires0={self.raw_wires[0].tolist()}, wires1={self.raw_wires[1].tolist()}, log_base={self.log_base})"
+
+    @property
+    def hash(self):
+        """int: returns an integer hash uniquely representing the measurement process"""
+        fingerprint = (
+            self.__class__.__name__,
+            tuple(self.raw_wires[0].tolist()),
+            tuple(self.raw_wires[1].tolist()),
+            self.log_base,
+        )
+
+        return hash(fingerprint)
+
+    @property
+    def return_type(self):
+        return VnEntanglementEntropy
+
+    @property
+    def numeric_type(self):
+        return float
+
+    def map_wires(self, wire_map: dict):
+        new_measurement = copy(self)
+        new_measurement._wires = [
+            Wires([wire_map.get(wire, wire) for wire in wires]) for wires in self.raw_wires
+        ]
+        return new_measurement
+
+    def shape(
+        self, shots: Optional[int] = None, num_device_wires: int = 0
+    ):  # pylint: disable=unused-argument
+        return ()
+
+    def process_state(self, state: Sequence[complex], wire_order: Wires):
+        state = qml.math.dm_from_state_vector(state)
+        return qml.math.vn_entanglement_entropy(
+            state,
+            indices0=list(self._wires[0]),
+            indices1=list(self._wires[1]),
+            c_dtype=state.dtype,
+            base=self.log_base,
+        )
```

Comment:
Can we implement `process_density_matrix` here too?

Reply:
Good idea!